### PR TITLE
Make Csrf cookie secure flag configurable (WebFlux)

### PR DESCRIPTION
**Pull request by ThomasVitale**
Make the XSRF-TOKEN cookie secure flag configurable in CookieServerCsrfTokenRepository.

Closes gh-9678

### CODE REVIEW

File: web/src/main/java/org/springframework/security/web/server/csrf/CookieServerCsrfTokenRepository.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2002-2019 the original author or authors.
+ * Copyright 2002-2020 the original author or authors.
```
Comment (eleftherias) [inline web/src/main/java/org/springframework/security/web/server/csrf/CookieServerCsrfTokenRepository.java:2]: 2021 :)

File: web/src/main/java/org/springframework/security/web/server/csrf/CookieServerCsrfTokenRepository.java [hunk 2]
```diff
@@ -146,6 +154,15 @@ public void setCookieDomain(String cookieDomain) {
 		this.cookieDomain = cookieDomain;
 	}
 
+	/**
+	 * Sets the cookie secure flag. If not set, the value depends on
+	 * {@link ServerHttpRequest#getSslInfo()}.
+	 * @param secure The value for the secure flag
+	 */
+	public void setSecure(boolean secure) {
```
Comment (eleftherias) [inline web/src/main/java/org/springframework/security/web/server/csrf/CookieServerCsrfTokenRepository.java:162]: Please add `@since 5.5`

**Review by eleftherias — CHANGES_REQUESTED** [review]
Thanks for the PR @ThomasVitale!
I've left a couple of comments.

File: web/src/main/java/org/springframework/security/web/server/csrf/CookieServerCsrfTokenRepository.java (hunk 1 above)
Comment (ThomasVitale) [reply web/src/main/java/org/springframework/security/web/server/csrf/CookieServerCsrfTokenRepository.java]: That's awkward :) I fixed it.

File: web/src/main/java/org/springframework/security/web/server/csrf/CookieServerCsrfTokenRepository.java (hunk 2 above)
Comment (ThomasVitale) [reply web/src/main/java/org/springframework/security/web/server/csrf/CookieServerCsrfTokenRepository.java:163]: Done

**Comment by ThomasVitale** [comment]
@eleftherias thanks for reviewing the PR. I have addressed your comments and pushed an update.

**Comment by eleftherias** [comment]
Thanks @ThomasVitale! This is now merged into main.

**Pull request merged**